projects/nx-verdaccio/src/plugin/caching.unit-test.ts

```diff
@@ -0,0 +1,168 @@
+      const recordToUpdate = { thunderfury: 'Soul of Sylvanas' };
+      setCacheRecord(targetsCache, prefix, hashData, cacheData);
+
+      expect(
+        setCacheRecord(targetsCache, prefix, hashData, recordToUpdate)
+      ).toBe(recordToUpdate);
+      expect(targetsCache).toHaveProperty(cacheKey, recordToUpdate);
+    });
+  });
+});
+
+describe('readTargetsCache', (): void => {
+  const path = 'azeroth';
+  let existsSyncSpy: MockInstance<[path: nodeFs.PathLike], boolean>;
+  let readJsonFileSpy: MockInstance<
+    [path: string, options?: JsonReadOptions],
+    object
+  >;
+
+  beforeEach((): void => {
+    existsSyncSpy = vi
+      .spyOn(nodeFs, 'existsSync')
+      .mockImplementation((): boolean => true);
+    readJsonFileSpy = vi
+      .spyOn(nxDevKit, 'readJsonFile')
+      .mockImplementation((): Record<string, Partial<ProjectConfiguration>> => {
+        return MOCK_TARGETS_CACHE;
+      });
+    vi.stubEnv("NX_CACHE_PROJECT_GRAPH", 'true')
+  });
+
+  afterEach((): void => {
+    existsSyncSpy.mockRestore();
+    readJsonFileSpy.mockRestore();
+    vi.clearAllMocks();
+  });
+
+  it('should call existSync once with correct argument', (): void => {
+    readTargetsCache(path);
+    expect(existsSyncSpy).toHaveBeenCalledWith(path);
+    expect(existsSyncSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call readJsonFile once with correct argument', (): void => {
+    readTargetsCache(path);
+    expect(readJsonFileSpy).toHaveBeenCalledWith(path);
+    expect(readJsonFileSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should return target cache if existsSync returns true, and NX_CACHE_PROJECT_GRAPH = true', (): void => {
+    expect(readTargetsCache(path)).toEqual(MOCK_TARGETS_CACHE);
+  });
+
+  it('should return empty object if NX_CACHE_PROJECT_GRAPH = false', (): void => {
+    process.env.NX_CACHE_PROJECT_GRAPH = 'false';
+    expect(readTargetsCache(path)).toEqual({});
+  });
+
+  it('should return empty object if existsSync returns false', (): void => {
+    existsSyncSpy.mockImplementation((): boolean => false);
+    expect(readTargetsCache(path)).toEqual({});
+  });
+
+  it('should return empty object if existsSync returns false, and NX_CACHE_PROJECT_GRAPH = false', (): void => {
+    existsSyncSpy.mockImplementation((): boolean => false);
+    process.env.NX_CACHE_PROJECT_GRAPH = 'false';
+    expect(readTargetsCache(path)).toEqual({});
+  });
+});
+
+describe('writeTargetsToCache', (): void => {
+  const writeJsonFile = vi
+    .spyOn(nxDevKit, 'writeJsonFile')
+    .mockImplementation((): string => 'dont write to file :D');
+  const path = 'azeroth';
+
+  afterEach((): void => {
+    writeJsonFile.mockRestore();
+    delete process.env.NX_CACHE_PROJECT_GRAPH;
+  });
+
+  it('should call writeJsonFile once with correct arguments if process.env.NX_CACHE_PROJECT_GRAPH !== false', (): void => {
+    process.env.NX_CACHE_PROJECT_GRAPH = 'true';
+    writeTargetsToCache(path, MOCK_TARGETS_CACHE);
```

Comment:
When working with Mocks is is good to try to keep them as small as possible and in best case colocate the data. In this case I guess a small object like "{prop: 42}" would also do the trick and will not require to open another file and look for the mock data